### PR TITLE
let huber loss return the same dtype as inputs

### PR DESCRIPTION
**Pull request by fsx950223**
I suggest changing https://github.com/tensorflow/tensorflow/blob/b3c4b50fa70868a2da709cb7868822e5361a50e4/tensorflow/python/keras/losses.py#L1529 to y_true.dtype too.
IMO:
1. It's annoyed when I run ```tf.keras.losses.categorical_crossentropy(tf.constant([[0, 1, 0], [0, 0, 1]], tf.float16), tf.constant([[0.05, 0.95, 0], [0.1, 0.8, 0.1]], tf.float16), label_smoothing=0.1)```, then I got dtype mismatch bug.
The result's dtype should depend on inputs.
2. Is it tf.keras.backend.set_xxx is going to be abandoned?

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR.

> Is it tf.keras.backend.set_xxx is going to be abandoned?

No, it is a current API.

All layers and losses generally cast their inputs to the global compute dtype when called. This mechanism enables mixed precision training, etc.

**Comment by fsx950223** [comment]
> Thanks for the PR.
> 
> > Is it tf.keras.backend.set_xxx is going to be abandoned?
> 
> No, it is a current API.
> 
> All layers and losses generally cast their inputs to the global compute dtype when called. This mechanism enables mixed precision training, etc.

I met the bug in custom train_step model and I reproduce it in follow example:
```python
import tensorflow as tf
tf.keras.mixed_precision.experimental.set_policy('float16')
print(tf.keras.backend.floatx())
print(tf.keras.losses.huber([1],[2],0.))
```
Does the behavior correct? @fchollet
No such note in https://www.tensorflow.org/guide/mixed_precision asks users to ```set_floatx```.
